main.py

```diff
@@ -33,11 +34,90 @@
 configs = {}
 device_db = {}
 
-mi_connection: Optional[CommMonInj] = None
-
 dataset_tracker: Optional[dset.DatasetTracker] = None
 schedule_tracker: Optional[schd.ScheduleTracker] = None
 
+def do_nothing(*_):
+    """Gets any input, but doesn't do anything."""
+
+
+class MonInj:
+    """Manages a connection to ARTIQ moninj proxy and each TTL status.
+    
+    Attributes:
+        connection: The CommMonInj object connecting to ARTIQ moninj proxy.
+        outputs, levels: The dictionary with each TTL status.
+          Each key is a TTL channel number and its value is a status object. Status means
+          the actual output value for outputs and the overriding level for levels.
+        overriding: The status object for overriding. It acts in common with all TTLs.
+    """
+    @dataclasses.dataclass
+    class Status:
+        """TTL status.
+        
+        Fields:
+            value: The current value.
+            modified: The asyncio event set once the value is updated.
+        """
+        value: Optional[bool] = None
+        modified: asyncio.Event = asyncio.Event()
```

Comment:
I did not intend it..
I think as we discussed, even if all the instances share it, it can detect a chance (although quite inefficient), hence it seemed to work well.